internal/config/options.go

```diff
@@ -349,3 +355,18 @@ func IsProxy(s string) bool {
 	}
 	return false
 }
+
+// OptionsUpdater updates local state based on an Options struct
+type OptionsUpdater interface {
+	UpdateOptions(*Options) error
+}
+
+// Checksum returns the checksum of the current options struct
+func (o *Options) Checksum() string {
+	hash, err := hashstructure.Hash(o, nil)
+
+	if err != nil {
```

Comment:
I debated a bit about what to do here since there is somehow a possibility of an error.  I think the safest thing to do is nothing but another option might be to return a value that will change once a minute or something along those lines.  

The only current use case would be running this _if_ the file changes but thinking further down the road (prometheus returning this, for instance) that might be a bad assumption.  Open to ideas.

Reply:
That's a really good point. I thought about this a bit as well and all the alternative methods (e.g. clock based or on some create date field) I can come up with feels like a path to unexpected edge cases.

I think you are right that the best thing to do is nothing; and if it turns out there are cases we didn't anticipate where the hashstruct fails, we can address it then. 

Maybe the one thing we could do is throw a warn log message.

Reply:
>Maybe the one thing we could do is throw a warn log message.

 Right; good call.  I look forward to figuring out the corner case that triggers it.